ahnlich/ai/src/tests/aiproxy_test.rs

```diff
@@ -623,6 +624,179 @@ async fn test_ai_proxy_get_sim_n_succeeds() {
+            }
+        } else if index_model_repr.input_type().as_str_name() == "IMAGE" {
+            StoreInput {
+                value: Some(Value::Image(include_bytes!("./images/dog.jpg").to_vec())),
+            }
+        } else {
+            StoreInput {
+                value: Some(Value::RawString("".into())),
+            }
+        };
+
+        let store_data = vec![
+            AiStoreEntry {
+                key: Some(store_input_1.clone()),
+                value: Some(store_value.clone()),
+            },
+            AiStoreEntry {
+                key: Some(store_input_2.clone()),
+                value: Some(store_value.clone()),
+            },
+        ];
+
+        // Create pipeline request
+        let queries = vec![
+            ai_pipeline::AiQuery {
+                query: Some(Query::CreateStore(ai_query_types::CreateStore {
+                    store: store_name.clone(),
+                    query_model: model,
+                    index_model: model,
+                    predicates: vec![],
+                    non_linear_indices: vec![],
+                    error_if_exists: true,
+                    store_original: true,
+                })),
+            },
+            ai_pipeline::AiQuery {
+                query: Some(Query::Set(ai_query_types::Set {
+                    store: store_name.clone(),
+                    inputs: store_data,
+                    preprocess_action: PreprocessAction::NoPreprocessing.into(),
+                    execution_provider: None,
+                })),
+            },
+        ];
+
+        let pipelined_request = ai_pipeline::AiRequestPipeline { queries };
+
+        let _ = client
+            .pipeline(tonic::Request::new(pipelined_request))
+            .await
+            .expect("Failed to send pipeline request");
+
+        let expected_entries = ai_response_types::StoreInputToEmbeddingsList {
+            values: vec![
+                SingleInputToEmbedding {
+                    input: Some(store_input_1.clone()),
+                    embedding: Some(StoreKey { key: vec![] }),
+                },
+                SingleInputToEmbedding {
+                    input: Some(store_input_2.clone()),
+                    embedding: Some(StoreKey { key: vec![] }),
+                },
+            ],
+        };
+
+        let store_inputs = vec![store_input_1, store_input_2];
+
+        let query = ai_query_types::ConvertStoreInputToEmbeddings {
+            store_inputs,
+            preprocess_action: Some(PreprocessAction::NoPreprocessing.into()),
+            model,
+        };
+
+        let response = client
+            .convert_store_input_to_embeddings(tonic::Request::new(query))
+            .await
+            .expect("Failed to convert store input to embeddings");
+
+        let response_entries = response.into_inner().values;
+
+        assert_eq!(response_entries.len(), expected_entries.values.len());
```

Comment:
This would then just be `assert_eq!(store_inputs.len(), response_entries.len())`